### PR TITLE
Adds hal+json, hal+xml, and ld+json #14796

### LINKED ISSUE #14796: RFC to add additional MIME types (HAL and JSON-LD) to Cake\HTTP\Response MIME types
This is a (multiple allowed):

* [ ] bug
* [ ] enhancement
* [X] feature-discussion (RFC)

Add HAL and JSON-LD mime types to [Response](https://github.com/cakephp/cakephp/blob/master/src/Http/Response.php).

Proposal:

```php
        'hal+json' => ['application/hal+json', 'application/vnd.hal+json'],
        'hal+xml' => ['application/hal+xml', 'application/vnd.hal+xml'],
        'ld+json' => 'application/ld+json',
```

I am not sure whether indexes for `hal+json` and `hal+xml` should instead by expressed as `vnd.hal+json` and `vnd.hal+xml`. There might be some debate there.

Not having these in necessitates some userland hacks to modify the Response at runtime such as including this code during the bootstrap phase of a plugin: https://gist.github.com/cnizzardini/1cf144d4d66b9f94f7d55b1c99d7ce62

Particularly, my use-case is building a HalJsonView and HalXmlView (and eventually a JsonLdView). It seems silly to have to add these MIME types in at runtime, though maybe there is a better way that I don't know about. Regardless, these are common enough message formats to be included IMO.

Reference:
- https://www.iana.org/assignments/media-types/media-types.xhtml
- http://stateless.co/hal_specification.html 
- https://json-ld.org/

### PR DESCRIPTION
**Pull request by cnizzardini**
Add hal+json, hal+xml, and ld+json #14796

I decided against modifying ServerRequest detectors, that can be handled by the user easily during bootstrap.

### CODE REVIEW

File: src/Http/Response.php [hunk 1]
```diff
@@ -161,6 +161,8 @@ class Response implements ResponseInterface
         'gz' => 'application/x-gzip',
         'bz2' => 'application/x-bzip',
         '7z' => 'application/x-7z-compressed',
+        'hal+json' => ['application/hal+json', 'application/vnd.hal+json'],
```
Comment (othercorey) [inline src/Http/Response.php:164]: Wouldn't this just be `hal` ?

File: src/Http/Response.php [hunk 2]
```diff
@@ -169,6 +171,7 @@ class Response implements ResponseInterface
         'js' => 'application/javascript',
         'jsonapi' => 'application/vnd.api+json',
         'latex' => 'application/x-latex',
+        'ld+json' => 'application/ld+json',
```
Comment (othercorey) [inline src/Http/Response.php:174]: Wouldn't this just be `jsonld` ?

File: src/Http/Response.php (hunk 1 above)
Comment (cnizzardini) [reply src/Http/Response.php:164]: There is a hal+xml specification too. How else should these be distinguished?

File: src/Http/Response.php (hunk 2 above)
Comment (cnizzardini) [reply src/Http/Response.php:174]: Yes, I think that makes more sense.

File: src/Http/Response.php (hunk 1 above)
Comment (othercorey) [reply src/Http/Response.php:164]: I never see `+` in a type name, but there doesn't seem to be any other name for hal objects. *shrug*

Comment (cnizzardini) [reply src/Http/Response.php:164]: Agreed, it's goofy. I am fine dropping hal+xml al together and just going with `'hal' => ['application/hal+json', 'application/vnd.hal+json']`.

**Comment by cnizzardini** [comment]
Were we done here?

**Comment by othercorey** [comment]
Not much we can do if we're supporting both. The name is what it is.

**Comment by markstory** [comment]
@cnizzardini If there is more to be done, more names can be added in additional pull requests.

**Pull request merged**